test/utility.cpp

```diff
@@ -91,6 +91,57 @@ TEST_CASE("exchange", "[utility.exchange]") {
   }
 }
 
+// [utility.as_const], as_const
+
+// template<class T>
+//   constexpr add_const_t<T>& as_const(T& t) noexcept;
+// template<class T>
+//   void as_const(const T&&) = delete;
+namespace as_const_deleted {
+void as_const(...);
+using slb::as_const;
+
+template <typename T>
+struct no_result_void {
+  using type = void;
+};
+
+template <typename T, typename Enable = void>
+struct no_result : std::true_type {};
+
+template <typename T>
+struct no_result<T,
```

Comment:
How about just this:
```
struct no_result<T, decltype(void(as_const(std::declval<T>())))>
    : std::false_type {};
```
and we don't need `no_result_void` here.

Reply:
We've gone in the opposite direction in the past, for readability.

Reply:
I assume you're referring to the one in `type_traits.cpp`, right?
```
// template <class Fn, class... ArgTypes> struct invoke_result;
template <typename T>
struct no_result_void {
  using type = void;
};

template <typename T, typename Enable = void>
struct no_result : std::true_type {};

template <typename T>
struct no_result<T, typename no_result_void<typename T::type>::type>
    : std::false_type {};
```

I guess the difference I had in my mind is that this was operating on a type,
whereas with `as_const` we're operating on an expression. Since with
an expression, we have a `decltype` in which we can stick a `void(...)`.

Anyway, perhaps this is too subtle of a difference to distinguish. I'm fine with
just keeping `no_result_void` as the pattern.